R/generics.R

```diff
@@ -651,6 +675,28 @@ setMethod("num_rep_cols", "MatrixPrintForm", function(obj) {
   }
 })
 
+#' @export
+#' @rdname num_rep_cols
+setMethod("num_rep_cols", "MatrixPrintForm", function(obj) obj$num_rep_cols)
```

Comment:
This needs to be merged with the above. Or this new value needs to be assigned with keycols in matrix_form from {rlistings}. If the second we need to change `.get_keycols_from_listing`

Reply:
We still need which keycols are there so I would keep the new entry in matrix_form. This would be probably a one-liner in matrix_form in rlistings (for num_rep_cols)

Reply:
I don't really understand this. It seems like keycolumns and rep_cols are mechanically the same, its just that one is specific to rlistings terminology (key column) while the other is general for anything that plugs into formatters.

The change here that I'm making afair is that instead of just automatically being calculated at toString/pagination time the number of rep-cols is carried around in the MPF now. Maybe I'm missing something else.

Reply:
It is probably going to work either way because num_rep_cols is assigned to the number of keycolumns. I think this may not be valid in rlistings hence my addition here, but I need to double-check. Either way is fine, we need to merge the two functions nonetheless as they work on the same object

Reply:
after this fix https://github.com/insightsengineering/rlistings/pull/211#discussion_r1600357912 we can remove the method strictly above this one

Reply:
It doesn't seem like this can be safely removed, this function (and the corresponding setter) are called in ~9 places in pagination.R...

At least sometimes its called on MatrixPrintForm objects (e.g. default value in `mf_infer_cinfo`)